libmachine/log/bugsnag.go

```diff
@@ -0,0 +1,54 @@
+package log
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/bugsnag/bugsnag-go"
+	"github.com/docker/machine/version"
+)
+
+var metaData bugsnag.MetaData
+
+func Setup(apiKey string) error {
+	if apiKey == "" {
+		apiKey = "e98db82ef639114001cd3397b2c9f885"
+	}
+	bugsnag.Configure(bugsnag.Configuration{
+		APIKey: apiKey,
+		// XXX we need to abuse bugsnag metrics to get the OS/ARCH information as a usable filter
+		// Can do that with either "stage" or "hostname"
+		ReleaseStage:    fmt.Sprintf("%s (%s)", runtime.GOOS, runtime.GOARCH),
+		ProjectPackages: []string{"github.com/docker/machine/[^v]*"},
+		AppVersion:      version.FullVersion(),
+		Synchronous:     true,
+		Logger:          new(BugsnagLogger),
+	})
+
+	metaData = bugsnag.MetaData{}
+
+	metaData.Add("app", "compiler", fmt.Sprintf("%s (%s)", runtime.Compiler, runtime.Version()))
+	metaData.Add("device", "os", runtime.GOOS)
+	metaData.Add("device", "arch", runtime.GOARCH)
+
+	// Add environment variables
```

Comment:
As mentioned in @dmp42 PR, we probably should not do this (report the user's environment variables) since it is highly likely to leak secrets -- for instance, the `DIGITALOCEAN_ACCESS_TOKEN` environment variable